mdata/store_cassandra.go

```diff
@@ -77,9 +78,37 @@ type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   map[uint32]string
 }
 
-func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, readers, writers, readqsize, writeqsize, retries, protoVer int, ssl, auth, hostVerification bool) (*cassandraStore, error) {
+func (c *cassandraStore) ensureTtlTables(ttls []uint32, keyspace string) error {
+	for _, ttl := range ttls {
+		// calculate table name and window
+		var window int = 1
+		for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
+			window *= 2
+		}
+		table_name := fmt.Sprintf("metric_%d", window)
+
+		// table isn't know to exist, ensure it exists
+		err := c.session.Query(fmt.Sprintf(table_schema, keyspace, table_name, window)).Exec()
```

Comment:
in the table in your comment there are multiple tables of the window size `1day`. Wouldn't it be better to put them together and use the window size as suffix of the table name? Otherwise what's the point of splitting them?